rdt-open-reader/src/test/java/io/ona/rdt/robolectric/util/UtilsTest.java

```diff
@@ -223,4 +230,43 @@ public void testIsValidJSONObjectShouldReturnCorrectStatus() {
         Assert.assertFalse(Utils.isValidJSONObject(null));
         Assert.assertFalse(Utils.isValidJSONObject(""));
     }
+
+    @Test
+    public void testVerifyUserAuthorizationShouldCallOnlyOnce() {
+
+        // verify pending state
+        Utils.UserAuthorizationVerificationTask userAuthorizationVerificationTask = Mockito.mock(Utils.UserAuthorizationVerificationTask.class);
+        ReflectionHelpers.setStaticField(Utils.UserAuthorizationVerificationTask.class, "INSTANCE", userAuthorizationVerificationTask);
+        Mockito.when(userAuthorizationVerificationTask.getStatus()).thenReturn(AsyncTask.Status.PENDING);
+        Utils.verifyUserAuthorization();
+        Mockito.verify(userAuthorizationVerificationTask, Mockito.times(1)).execute();
+
+        // verify finished state
+        Mockito.when(userAuthorizationVerificationTask.getStatus()).thenReturn(AsyncTask.Status.FINISHED);
+        Utils.verifyUserAuthorization();
+        Mockito.verify(userAuthorizationVerificationTask, Mockito.times(1)).destroyInstance();
+        Mockito.verify(userAuthorizationVerificationTask, Mockito.times(2)).execute();
+        ReflectionHelpers.setStaticField(Utils.UserAuthorizationVerificationTask.class, "INSTANCE", null);
+    }
+
+    @Test
+    public void TestUserAuthorizationVerificationTaskShouldVerifyAllCases() throws AuthenticatorException, OperationCanceledException, IOException {
```

Comment:
Not sure, what use case that this test?

Reply:
verify the normal and exception case in task

Reply:
doesn't the test case above verify the normal flows?

Also for exceptions, simply add the `@test(expected)` annotation, no need to mock Timber.

Reply:
we can't use `@test(expected)` because we catch the exception inside try-catch

Reply:
if this only tests an exception is caught then the test is not necessary, we can remove it

Reply:
test removed.